### ISSUE TITLE
Control direction of dropdowns

### ISSUE THREAD

**Issue by Bokelmann** [gantry/gantry5#1058]
I would like to have more control about the direction of dropdowns (like we had in Gantry4).
I haven't really thought about the the 1. leve-dropdowns. But for deeper levels I could think of something like ...

@import "dependencies";

@mixin openleft {
left: auto!important;
right: 100%!important
}

@mixin openright {
left: 100%!important;
right: auto!important
}

@include breakpoint(large-desktop-range) {
.drop-left-large-desktop > ul {@include openleft;}
.drop-right-large-desktop > ul {@include openright;}
}

@include breakpoint(desktop-range) {
.drop-left-desktop > ul {@include openleft;}
.drop-right-desktop > ul {@include openright;}
}

@include breakpoint(t-range) {
.drop-left-tablet > ul {@include openleft;}
.drop-right-tablet > ul {@include openright;}
}

**Comment by JoomFX** [comment]
Hey @Bokelmann ,

I just provide a utility class with my templates called `flyout-left`:

```
// Flyout on the Left Utility
.g-main-nav {
	.g-standard {
		.g-dropdown {
			.flyout-left {
				.g-dropdown {
					left: auto;
					right: 100%;
				}
			}
		}
	}
}
```

Then the user just adds the class in the Menu Editor, to the menu items they want to change the flyout position for:

![alt](http://i.imgur.com/lLuDAhB.png)

It will be nice if such utility class is added to Nucleus... I'm not sure if it will be better to have such an option (switch for left and right) in the GUI, in the Menu Editor or just the utility class.

**Comment by mahagr** [comment]
@w00fz This is more of your area, what do you think?

**Comment by Bokelmann** [comment]
what @JoomFX Ivo provided is more or less the same as my idea.
I also would like to have control for the direction across the given viewports.
Just a bit more fine grained ...

**Comment by w00fz** [comment]
I'm totally :ok_hand: for a pure CSS solution.
I think it would be nicer if it was a field (checkbox or dropdown) that applied a `.g-dropdown-left` or `.g-dropdown-right` [default] class to the item. 

From the CSS standpoint, I like it being a mixin. This will make it work out of the box in an easy and reachable way for users, by just selecting the direction from a dropdown and without having to guess classnames. You developers will have access to the mixin and can easily override its behavior or extend per breakpoint in case needed, instead leveraging the load on Gantry by adding the different breakpoints classes.

So to recap: 

1. All items will have either `.g-dropdown-left` or `.g-dropdown-right` [by default all will have `.g-dropdown-right`]
2. Two new mixins:
  ``` sass
  @mixin dropdown-left {
    left: auto;
    right: 100%;
  }
  
  @mixin dropdown-right {
    left: 100%;
    right: auto;
  }
  
  .g-dropdown-left { @include dropdown-left; }
  .g-dropdown-right { @include dropdown-right; }
  ```
3. All Menu Items in Menu Manager will get a new dropdown field `Dropdown Direction` with 2 options:
  
  I. Expand Left
  II. Expand Right _[default]_

I think this would cover all needs. If you need fine grained customizations, like for breakpoints, you have access to the mixin. If you know nothing about CSS, you can easily switch from the admin UI. If you need to do crazy stuff from the twig, you have access to the option.

What do you guys think?

**Comment by JoomFX** [comment]
> What do you guys think?

Just perfect :+1: 
I'm fully satisfied with the above :smile:

**Comment by Bokelmann** [comment]
Good enough for me.

**Comment by ramon12** [comment]
Should add the middle one too

**Comment by w00fz** [comment]
Wouldn't it overlap the menu item if it was in the middle?

**Comment by ramon12** [comment]
No, here live test ![Middle](http://s28.postimg.org/qs9sp5avx/middle.png)

**Comment by JoomFX** [comment]
@ramon12 As far as I understand from your screenshot, you are talking about the second level menu (the first dropdown), while we are discussing the 3rd/4th... levels, the flyout menus.

I see what you mean but I do not think it should be made a core G5 feature. This can be easily achieved with couple of lines of CSS in the `custom.scss` file.

Correct me if I have misunderstood you.

**Comment by Bokelmann** [comment]
hmmm I actually got a requests for that centered 2nd-level dropdown in the past. And yes that's easy to add via CSS. Not sure. But that 3rd (dropcenter) option would it make more complete me thinks ;-)
Thanks for adding feedback guys!

**Comment by ramon12** [comment]
@JoomFX It doesn't matter what menu level are you, first,second or last, the suggestion is to make the direction of dropdown, see other frameworks they have this option. In the image i was using custom particle, if you want i can provide other examples if you want.

**Comment by JoomFX** [comment]
Alright guys, I see...

So it is @w00fz turn now to decide if this should be implemented or not.

**Comment by w00fz** [comment]
This is now implemented and I also added the center option, however the center makes sense only at the 1st level, below that it will overlap and look "broken". This is now going to be considered an intentional behavior and we won't provide special styling for it, we will rely on the common sense on your part for its usage and/or ability to style it accordingly so that it's usable :wink: 

Aside from that, I think this is a great addition to the menu, thank you guys for the suggestion!

If you guys want to give it a try, that would be awesome. It should be soon available with 1b88f11
